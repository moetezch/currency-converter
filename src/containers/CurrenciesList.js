import React, { Component } from 'react';
import currenciesListArray from '../utils/currenciesListArray'

class CurrenciesList extends Component {
  // default list of currencies
  state = { currencies: ["IDR", "EUR", "GBP", "SGD"] }
  // Adding another currency to the list
  onCurrencyAdd = event => {
    let value = event.target.value
    if (!this.state.currencies.includes(value)) {
      this.setState(() => ({ currencies: [...this.state.currencies, value] }));

    }
    event.target.value = "add"
  };
  // Showing the list of converted currencies 
  filterCurrencies(filtered, rates, amount, selectedCurrency) {
    return Object.entries(rates).map(([currency, rate]) => {
      return filtered.map((filteredcurrency) => {
        if (filteredcurrency === currency) {
          return (
            <div className="alert alert-light alert-dismissible fade show" role="alert" key={currency}>
              <p>{currency} {(rate * amount).toFixed(4)}</p>
              <p>{currency}-{(currenciesListArray.find((curr) => {
                return curr.includes(currency)
              }))[1]}
              </p>
              <p>1 {selectedCurrency} = {rate.toFixed(4)} {currency}</p>
              <button type="button" className="close" data-dismiss="alert" aria-label="Close" onClick={() => {
                const array = this.state.currencies
                array.splice(this.state.currencies.indexOf(currency), 1)
                this.setState(() => ({ currencies: array }));
              }}>
                <span aria-hidden="true">&times;</span>
              </button>
            </div>

          )
        }
      })
    })
  }
  render() {
    const { rates, amount, selectedCurrency } = this.props
    return (
      <div>
        {this.filterCurrencies(this.state.currencies, rates, amount, selectedCurrency)}
        <div className="row">
          <div className="col">
            <select className="custom-select form-control" onChange={this.onCurrencyAdd}>
              <option value="add">Add another Currency</option>
              {currenciesListArray.map((currency) => {
                return (<option value={currency[0]} key={currency[0]}>{currency[0]}</option>)
              })}
            </select>
          </div>
        </div>
      </div>
    );
  }
}


export default CurrenciesList;