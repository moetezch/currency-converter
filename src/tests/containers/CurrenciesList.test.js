import {shallow} from 'enzyme'
import React from 'react'
import CurrenciesList from '../../containers/CurrenciesList'

it('expect to render currencies list',()=>{
const mockselectedCurrency="USD"
const mockRates={
  "CAD": 1.565,
  "CHF": 1.1798,
}
const mockAmount=20
  const wrapper = shallow(<CurrenciesList rates={mockRates} amount={mockAmount} selectedCurrency={mockselectedCurrency}/>)
  expect (wrapper).toMatchSnapshot()
})