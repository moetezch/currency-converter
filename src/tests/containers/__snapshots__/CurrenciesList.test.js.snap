// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render currencies list 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrenciesList
    amount={20}
    rates={
        Object {
            "CAD": 1.565,
            "CHF": 1.1798,
          }
    }
    selectedCurrency="USD"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
          ],
        ],
        <div
          className="row"
>
          <div
                    className="col"
          >
                    <select
                              className="custom-select form-control"
                              onChange={[Function]}
                    >
                              <option
                                        value="add"
                              >
                                        Add another Currency
                              </option>
                              <option
                                        value="USD"
                              >
                                        USD
                              </option>
                              <option
                                        value="EUR"
                              >
                                        EUR
                              </option>
                              <option
                                        value="CAD"
                              >
                                        CAD
                              </option>
                              <option
                                        value="IDR"
                              >
                                        IDR
                              </option>
                              <option
                                        value="GBP"
                              >
                                        GBP
                              </option>
                              <option
                                        value="CHF"
                              >
                                        CHF
                              </option>
                              <option
                                        value="SGD"
                              >
                                        SGD
                              </option>
                              <option
                                        value="INR"
                              >
                                        INR
                              </option>
                              <option
                                        value="MYR"
                              >
                                        MYR
                              </option>
                              <option
                                        value="JPY"
                              >
                                        JPY
                              </option>
                              <option
                                        value="KRW"
                              >
                                        KRW
                              </option>
                    </select>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col"
>
            <select
                        className="custom-select form-control"
                        onChange={[Function]}
            >
                        <option
                                    value="add"
                        >
                                    Add another Currency
                        </option>
                        <option
                                    value="USD"
                        >
                                    USD
                        </option>
                        <option
                                    value="EUR"
                        >
                                    EUR
                        </option>
                        <option
                                    value="CAD"
                        >
                                    CAD
                        </option>
                        <option
                                    value="IDR"
                        >
                                    IDR
                        </option>
                        <option
                                    value="GBP"
                        >
                                    GBP
                        </option>
                        <option
                                    value="CHF"
                        >
                                    CHF
                        </option>
                        <option
                                    value="SGD"
                        >
                                    SGD
                        </option>
                        <option
                                    value="INR"
                        >
                                    INR
                        </option>
                        <option
                                    value="MYR"
                        >
                                    MYR
                        </option>
                        <option
                                    value="JPY"
                        >
                                    JPY
                        </option>
                        <option
                                    value="KRW"
                        >
                                    KRW
                        </option>
            </select>
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <select
              className="custom-select form-control"
              onChange={[Function]}
>
              <option
                            value="add"
              >
                            Add another Currency
              </option>
              <option
                            value="USD"
              >
                            USD
              </option>
              <option
                            value="EUR"
              >
                            EUR
              </option>
              <option
                            value="CAD"
              >
                            CAD
              </option>
              <option
                            value="IDR"
              >
                            IDR
              </option>
              <option
                            value="GBP"
              >
                            GBP
              </option>
              <option
                            value="CHF"
              >
                            CHF
              </option>
              <option
                            value="SGD"
              >
                            SGD
              </option>
              <option
                            value="INR"
              >
                            INR
              </option>
              <option
                            value="MYR"
              >
                            MYR
              </option>
              <option
                            value="JPY"
              >
                            JPY
              </option>
              <option
                            value="KRW"
              >
                            KRW
              </option>
</select>,
            "className": "col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="add"
>
                  Add another Currency
</option>,
                Array [
                  <option
                    value="USD"
>
                    USD
</option>,
                  <option
                    value="EUR"
>
                    EUR
</option>,
                  <option
                    value="CAD"
>
                    CAD
</option>,
                  <option
                    value="IDR"
>
                    IDR
</option>,
                  <option
                    value="GBP"
>
                    GBP
</option>,
                  <option
                    value="CHF"
>
                    CHF
</option>,
                  <option
                    value="SGD"
>
                    SGD
</option>,
                  <option
                    value="INR"
>
                    INR
</option>,
                  <option
                    value="MYR"
>
                    MYR
</option>,
                  <option
                    value="JPY"
>
                    JPY
</option>,
                  <option
                    value="KRW"
>
                    KRW
</option>,
                ],
              ],
              "className": "custom-select form-control",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add another Currency",
                  "value": "add",
                },
                "ref": null,
                "rendered": "Add another Currency",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "USD",
                "nodeType": "host",
                "props": Object {
                  "children": "USD",
                  "value": "USD",
                },
                "ref": null,
                "rendered": "USD",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "EUR",
                "nodeType": "host",
                "props": Object {
                  "children": "EUR",
                  "value": "EUR",
                },
                "ref": null,
                "rendered": "EUR",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "CAD",
                "nodeType": "host",
                "props": Object {
                  "children": "CAD",
                  "value": "CAD",
                },
                "ref": null,
                "rendered": "CAD",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "IDR",
                "nodeType": "host",
                "props": Object {
                  "children": "IDR",
                  "value": "IDR",
                },
                "ref": null,
                "rendered": "IDR",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "GBP",
                "nodeType": "host",
                "props": Object {
                  "children": "GBP",
                  "value": "GBP",
                },
                "ref": null,
                "rendered": "GBP",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "CHF",
                "nodeType": "host",
                "props": Object {
                  "children": "CHF",
                  "value": "CHF",
                },
                "ref": null,
                "rendered": "CHF",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "SGD",
                "nodeType": "host",
                "props": Object {
                  "children": "SGD",
                  "value": "SGD",
                },
                "ref": null,
                "rendered": "SGD",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "INR",
                "nodeType": "host",
                "props": Object {
                  "children": "INR",
                  "value": "INR",
                },
                "ref": null,
                "rendered": "INR",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "MYR",
                "nodeType": "host",
                "props": Object {
                  "children": "MYR",
                  "value": "MYR",
                },
                "ref": null,
                "rendered": "MYR",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "JPY",
                "nodeType": "host",
                "props": Object {
                  "children": "JPY",
                  "value": "JPY",
                },
                "ref": null,
                "rendered": "JPY",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "KRW",
                "nodeType": "host",
                "props": Object {
                  "children": "KRW",
                  "value": "KRW",
                },
                "ref": null,
                "rendered": "KRW",
                "type": "option",
              },
            ],
            "type": "select",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
            ],
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ],
          <div
            className="row"
>
            <div
                        className="col"
            >
                        <select
                                    className="custom-select form-control"
                                    onChange={[Function]}
                        >
                                    <option
                                                value="add"
                                    >
                                                Add another Currency
                                    </option>
                                    <option
                                                value="USD"
                                    >
                                                USD
                                    </option>
                                    <option
                                                value="EUR"
                                    >
                                                EUR
                                    </option>
                                    <option
                                                value="CAD"
                                    >
                                                CAD
                                    </option>
                                    <option
                                                value="IDR"
                                    >
                                                IDR
                                    </option>
                                    <option
                                                value="GBP"
                                    >
                                                GBP
                                    </option>
                                    <option
                                                value="CHF"
                                    >
                                                CHF
                                    </option>
                                    <option
                                                value="SGD"
                                    >
                                                SGD
                                    </option>
                                    <option
                                                value="INR"
                                    >
                                                INR
                                    </option>
                                    <option
                                                value="MYR"
                                    >
                                                MYR
                                    </option>
                                    <option
                                                value="JPY"
                                    >
                                                JPY
                                    </option>
                                    <option
                                                value="KRW"
                                    >
                                                KRW
                                    </option>
                        </select>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col"
>
              <select
                            className="custom-select form-control"
                            onChange={[Function]}
              >
                            <option
                                          value="add"
                            >
                                          Add another Currency
                            </option>
                            <option
                                          value="USD"
                            >
                                          USD
                            </option>
                            <option
                                          value="EUR"
                            >
                                          EUR
                            </option>
                            <option
                                          value="CAD"
                            >
                                          CAD
                            </option>
                            <option
                                          value="IDR"
                            >
                                          IDR
                            </option>
                            <option
                                          value="GBP"
                            >
                                          GBP
                            </option>
                            <option
                                          value="CHF"
                            >
                                          CHF
                            </option>
                            <option
                                          value="SGD"
                            >
                                          SGD
                            </option>
                            <option
                                          value="INR"
                            >
                                          INR
                            </option>
                            <option
                                          value="MYR"
                            >
                                          MYR
                            </option>
                            <option
                                          value="JPY"
                            >
                                          JPY
                            </option>
                            <option
                                          value="KRW"
                            >
                                          KRW
                            </option>
              </select>
</div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                className="custom-select form-control"
                onChange={[Function]}
>
                <option
                                value="add"
                >
                                Add another Currency
                </option>
                <option
                                value="USD"
                >
                                USD
                </option>
                <option
                                value="EUR"
                >
                                EUR
                </option>
                <option
                                value="CAD"
                >
                                CAD
                </option>
                <option
                                value="IDR"
                >
                                IDR
                </option>
                <option
                                value="GBP"
                >
                                GBP
                </option>
                <option
                                value="CHF"
                >
                                CHF
                </option>
                <option
                                value="SGD"
                >
                                SGD
                </option>
                <option
                                value="INR"
                >
                                INR
                </option>
                <option
                                value="MYR"
                >
                                MYR
                </option>
                <option
                                value="JPY"
                >
                                JPY
                </option>
                <option
                                value="KRW"
                >
                                KRW
                </option>
</select>,
              "className": "col",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="add"
>
                    Add another Currency
</option>,
                  Array [
                    <option
                      value="USD"
>
                      USD
</option>,
                    <option
                      value="EUR"
>
                      EUR
</option>,
                    <option
                      value="CAD"
>
                      CAD
</option>,
                    <option
                      value="IDR"
>
                      IDR
</option>,
                    <option
                      value="GBP"
>
                      GBP
</option>,
                    <option
                      value="CHF"
>
                      CHF
</option>,
                    <option
                      value="SGD"
>
                      SGD
</option>,
                    <option
                      value="INR"
>
                      INR
</option>,
                    <option
                      value="MYR"
>
                      MYR
</option>,
                    <option
                      value="JPY"
>
                      JPY
</option>,
                    <option
                      value="KRW"
>
                      KRW
</option>,
                  ],
                ],
                "className": "custom-select form-control",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add another Currency",
                    "value": "add",
                  },
                  "ref": null,
                  "rendered": "Add another Currency",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "USD",
                  "nodeType": "host",
                  "props": Object {
                    "children": "USD",
                    "value": "USD",
                  },
                  "ref": null,
                  "rendered": "USD",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "EUR",
                  "nodeType": "host",
                  "props": Object {
                    "children": "EUR",
                    "value": "EUR",
                  },
                  "ref": null,
                  "rendered": "EUR",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "CAD",
                  "nodeType": "host",
                  "props": Object {
                    "children": "CAD",
                    "value": "CAD",
                  },
                  "ref": null,
                  "rendered": "CAD",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "IDR",
                  "nodeType": "host",
                  "props": Object {
                    "children": "IDR",
                    "value": "IDR",
                  },
                  "ref": null,
                  "rendered": "IDR",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "GBP",
                  "nodeType": "host",
                  "props": Object {
                    "children": "GBP",
                    "value": "GBP",
                  },
                  "ref": null,
                  "rendered": "GBP",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "CHF",
                  "nodeType": "host",
                  "props": Object {
                    "children": "CHF",
                    "value": "CHF",
                  },
                  "ref": null,
                  "rendered": "CHF",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "SGD",
                  "nodeType": "host",
                  "props": Object {
                    "children": "SGD",
                    "value": "SGD",
                  },
                  "ref": null,
                  "rendered": "SGD",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "INR",
                  "nodeType": "host",
                  "props": Object {
                    "children": "INR",
                    "value": "INR",
                  },
                  "ref": null,
                  "rendered": "INR",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "MYR",
                  "nodeType": "host",
                  "props": Object {
                    "children": "MYR",
                    "value": "MYR",
                  },
                  "ref": null,
                  "rendered": "MYR",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "JPY",
                  "nodeType": "host",
                  "props": Object {
                    "children": "JPY",
                    "value": "JPY",
                  },
                  "ref": null,
                  "rendered": "JPY",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "KRW",
                  "nodeType": "host",
                  "props": Object {
                    "children": "KRW",
                    "value": "KRW",
                  },
                  "ref": null,
                  "rendered": "KRW",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
